### PR TITLE
adding a new argument on \Backend\Shared::start for specific FastCGI Params

### PR DESCRIPTION
**Pull request by Pierozi**
This PR is in relation with Issue #2

The new parameter are named **$fcgiParams**, with him, we can specify more parameters dedicated to `\Hoa\Fastcgi\Responder`.

It's useful for specify parameters dedicated to *Worker*, can read in *Super Global* **$_SERVER**.

**An Exception will be triggered when variable are different than array.**

### CODE REVIEW

**Comment by Pierozi** [comment]
@Hywan ask for review.

File: Backend/Shared.php [hunk 1]
```diff
@@ -314,24 +314,31 @@ public function run ( ) {
...
-     * @param   string  $socket        Socket URI to PHP-FPM server.
-     * @param   string  $workerPath    Path to the shared worker program.
+     * @param   string  $socket             Socket URI to PHP-FPM server.
+     * @param   string  $workerPath         Path to the shared worker program.
+     * @param   array   $fastcgiParameters  Array of additional parameters for FastCGI.
+     * @throw  \Hoa\Worker\Backend\Exception
      * @return  bool
      */
-    public static function start ( $socket, $workerPath ) {
+    public static function start ( $socket, $workerPath, array $fastcgiParameters = array() ) {
```
Comment (Hywan) [inline Backend/Shared.php]: We type-hint with `Array`.

File: Backend/Shared.php [hunk 2]
```diff
@@ -314,24 +314,31 @@ public function run ( ) {
...
 
         $server = new \Hoa\Fastcgi\Responder(
             new \Hoa\Socket\Client($socket)
         );
 
-        return $server->send(array(
+        $headers = [
```
Comment (Hywan) [inline Backend/Shared.php]: Use `array(` instead of `[` if the package has not dropped PHP5.3 yet.

File: Backend/Shared.php [hunk 3]
```diff
@@ -314,24 +314,31 @@ public function run ( ) {
...
             'SERVER_PROTOCOL'   => 'HTTP/1.1',
-            'REQUEST_METHOD'    => 'GET',
             'REQUEST_URI'       => $workerPath,
             'SCRIPT_FILENAME'   => $workerPath,
             'SCRIPT_NAME'       => DS . dirname($workerPath)
-        ));
+        ];
+
+        $fastcgiParameters = array_merge([
+            'REQUEST_METHOD'    => 'GET'
```
Comment (Hywan) [inline Backend/Shared.php]: Remove useless spaces before `=>`.

File: Backend/Shared.php [hunk 4]
```diff
@@ -314,24 +314,31 @@ public function run ( ) {
...
+        ], $fastcgiParameters);
+
+        return $server->send(array_merge($fastcgiParameters, $headers));
```
Comment (Hywan) [inline Backend/Shared.php]: We do 2 `array_merge`. Could we make only one?

Comment (Pierozi) [inline Backend/Shared.php]: you right, we can do `array_merge(['REQUEST_METHOD' => 'GET'], $fastcgiParameters, $headers)`

File: Backend/Shared.php (hunk 2 above)
Comment (Pierozi) [inline Backend/Shared.php]: Ok, once this PR done, I can do PR for drop PHP 5.3

Comment (Hywan) [inline Backend/Shared.php]: Good idea.

**Comment by Pierozi** [comment]
@Hywan review ?

**Comment by Pierozi** [comment]
Review @hoaproject/hoackers

**Comment by Hywan** [comment]
:+1:

**Comment by Hywan** [comment]
Thanks!

**Pull request merged**